frontend/src/scenes/insights/insightDataTimingLogic.ts

```diff
@@ -0,0 +1,116 @@
+import { dataNodeLogic } from '~/queries/nodes/DataNode/dataNodeLogic'
+import { insightVizDataNodeKey } from '~/queries/nodes/InsightViz/InsightViz'
+import { DataNode } from '~/queries/schema'
+import { InsightLogicProps } from '~/types'
+import { keyForInsightLogicProps } from './sharedUtils'
+
+import type { insightDataTimingLogicType } from './insightDataTimingLogicType'
+import { teamLogic } from 'scenes/teamLogic'
+import { captureTimeToSeeData } from 'lib/internalMetrics'
+
+export const insightDataTimingLogic = kea<insightDataTimingLogicType>([
+    props({} as InsightLogicProps),
+    key(keyForInsightLogicProps('new')),
+    path((key) => ['scenes', 'insights', 'insightDataTimingLogic', key]),
+    connect((props: InsightLogicProps) => ({
+        values: [
+            teamLogic,
+            ['currentTeamId'],
+            dataNodeLogic({ key: insightVizDataNodeKey(props), query: {} as DataNode }),
+            ['query', 'response'],
+        ],
+        actions: [
+            // TODO: need to pass empty query here, as otherwise dataNodeLogic will throw
+            dataNodeLogic({ key: insightVizDataNodeKey(props), query: {} as DataNode }),
+            ['loadData', 'loadDataSuccess', 'loadDataFailure', 'abortQuery as loadDataCancellation'],
+        ],
+    })),
+    actions({
+        startQuery: (queryId: string) => ({ queryId }),
+        removeQuery: (queryId: string) => ({ queryId }),
+    }),
+    reducers({
+        queryStartTimes: [
+            {} as Record<string, number>,
+            {
+                startQuery: (state, { queryId }) => ({ ...state, [queryId]: performance.now() }),
+                removeQuery: (state, { queryId }) => {
+                    const { [queryId]: remove, ...rest } = state
+                    return rest
+                },
+            },
+        ],
+    }),
+    listeners(({ actions, values }) => ({
+        loadData: ({ queryId }) => {
+            console.debug('loadData', queryId)
+            actions.startQuery(queryId)
+        },
+        loadDataSuccess: ({ payload }) => {
+            // ignore initialization
+            if (!payload || !values.queryStartTimes[payload.queryId]) {
+                return
+            }
+
+            const duration = performance.now() - values.queryStartTimes[payload.queryId]
+
+            captureTimeToSeeData(values.currentTeamId, {
+                type: 'insight_load',
+                context: 'insight',
+                primary_interaction_id: payload.queryId,
+                query_id: payload.queryId,
+                status: 'success',
+                time_to_see_data_ms: Math.floor(duration),
+                insights_fetched: 1,
+                insights_fetched_cached: values.response?.is_cached ? 1 : 0,
+                // api_response_bytes: values.response?.apiResponseBytes, getResponseB
+                // api_url: values.response?.apiUrl,
+                insight: values.query.kind,
```

Comment:
This is now using `query.kind` e.g. "TrendsQuery" instead of "TRENDS", "FUNNELS", etc. to support other query types.